cirq/circuits/expand_composite.py

```diff
@@ -54,12 +54,4 @@ def optimization_at(self, circuit, index, op):
             new_operations=decomposition)
 
     def _decompose(self, op: ops.Operation) -> ops.OP_TREE:
-        """Recursively decompose composite gates into an OP_TREE of gates."""
-        skip = self.no_decomp(op)
-        if skip and (skip is not NotImplemented):
-            return op
-
-        decomposed = protocols.decompose_once(op, None)
-        if decomposed is None:
-            return op
-        return (self._decompose(op) for op in decomposed)
+        return protocols.decompose(op, keep=self.no_decomp, on_stuck_raise=None)
```

Comment:
This function is so simple now that it should be inlined into the caller and then deleted.